pennylane/tape/tape.py

```diff
@@ -400,6 +400,7 @@ def _update_circuit_info(self):
         )
         self.num_wires = len(self.wires)
         self.all_sampled = all(m.return_type is Sample for m in self.measurements)
+        self.is_sampled = any(m.return_type is Sample for m in self.measurements)
```

Comment:
I think this is a bug in master! This line is needed to update `is_sampled` when the tape is expanded, because in the background a new tape is created.

Reply:
Good catch. It seems like this should be here, I agree. Although `is_sampled` seems to be be update in `_process_queue` as well.